depends/thirdparty/orc/tools/src/HdfsFileContents.cc

```diff
@@ -0,0 +1,101 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdlib>
+#include <memory>
+#include <string>
+#include <iostream>
+#include <string>
+#include <stdexcept>
+
+#include "orc/orc-config.hh"
+#include "orc/ColumnPrinter.hh"
+
+#include "wrap/libhdfs3-wrapper.h"
+
+void printContents(const char* filename, const orc::ReaderOptions opts) {
+  std::unique_ptr<orc::Reader> reader;
+
+  auto addr = std::getenv("LIBHDFS3_NAMENODE_ADDRESS");
+  std::string nn = "localhost";
+  if(addr != NULL) {
+    nn = std::string(addr);
+  }
+  auto portStr = std::getenv("LIBHDFS3_NAMENODE_PORT");
+  tPort port = 8020;
+  if (portStr != NULL) {
+    port = static_cast<tPort>(atoi(portStr));
+  }
+  hdfsFS fs = hdfsConnect(nn.c_str(), port);
+  reader = orc::createReader(orc::readHdfsFile(fs, std::string(filename)), opts);
+
+  std::unique_ptr<orc::ColumnVectorBatch> batch = reader->createRowBatch(1000);
```

Comment:
Can we extract "1000" out to a input parameter?  I think it will be very useful if the utility can help print some information depending on input.